orttraining/orttraining/python/training/_module_output_transformation.py

```diff
@@ -0,0 +1,141 @@
+import torch
+from collections import abc, OrderedDict
+
+def _process_output_name(output_name, output_index = None):
+    processed_name = output_name.replace('*', '**')
+    return processed_name if output_index is None else processed_name + '*{}'.format(output_index)
+
+def _unprocess_output_name(processed_name):
+    index_appended = processed_name.count('*') % 2 == 1
+    output_index = None
+    if index_appended:
+        output_index = int(processed_name[processed_name.rfind('*')+1:])
+        processed_name = processed_name[:processed_name.rfind('*')]
+    return processed_name.replace('**', '*'), output_index
```

Comment:
How does the output names changed from current implementation?
Why it was needed?
Why replace * by ** and add another *{} when an index is provided?